pkg/apis/componentconfig/v1alpha1/defaults.go

```diff
@@ -95,6 +95,14 @@ func SetDefaults_KubeControllerManagerConfiguration(obj *KubeControllerManagerCo
 	if obj.HPAController.HorizontalPodAutoscalerUpscaleForbiddenWindow == zero {
 		obj.HPAController.HorizontalPodAutoscalerUpscaleForbiddenWindow = metav1.Duration{Duration: 3 * time.Minute}
 	}
+	if obj.HPAController.HorizontalPodAutoscalerCPUTaintPeriod == zero {
+		// Assuming CPU is collected every minute and initialization takes another minute HPA should
+		// disregard samples from first two minutes as contaminated by initialization.
+		obj.HPAController.HorizontalPodAutoscalerCPUTaintPeriod = metav1.Duration{Duration: time.Minute}
+	}
+	if obj.HPAController.HorizontalPodAutoscalerInitialReadinessDelay == zero {
+		obj.HPAController.HorizontalPodAutoscalerInitialReadinessDelay = metav1.Duration{Duration: 10 * time.Second}
```

Comment:
10 sec only?

Reply:
Yes. This is what we assume is max duration between pod starting and the first time it get readiness assigned

Reply:
It was <2s for the pods I looked at

Reply:
Let's make it 30second, just in case Kubelet is stalled.